frontend/src/types/sheet.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  SheetId,
+  Instance,
+  InstanceId,
+  Database,
+  DatabaseId,
+  Principal,
+} from ".";
+
+export type SheetVisibility = "PRIVATE" | "PROJECT" | "PUBLIC";
+
+export interface Sheet {
+  id: SheetId;
+
+  // Standard fields
+  creator: Principal;
+  createdTs: number;
+  updater: Principal;
+  updatedTs: number;
+
+  // Related fields
+  instanceId: InstanceId;
+  instance: Instance;
+  databaseId?: DatabaseId;
+  database?: Database;
+
+  // Domain fields
+  name: string;
+  statement: string;
+  visibility: SheetVisibility;
+}
+
+export type CreateSheetState = Omit<
+  Sheet,
+  "id" | "creator" | "createdTs" | "updater" | "updatedTs"
+>;
+
+export type SheetPatch = Partial<
+  Pick<Sheet, "id" | "name" | "statement" | "visibility">
+>;
+
+export type AccessOption = {
+  label: string;
```

Comment:
should label be changed to name?

Reply:
Rename to SharingOption ?

Reply:
In the frontend UI， render some list, the option type usually looks like this:

```ts
type option = {
    label: string,
    value: any,
}
````

Why do we have to use `name` as the key?

some demo:

* https://element.eleme.cn/#/zh-CN/component/select#option-attributes
* https://www.naiveui.com/zh-CN/os-theme/components/select
* https://ant.design/components/select-cn/#OptGroup-props